File: ComplexNumProjJava/Complex.java
public class Complex {

    public Complex(int norm) {

        this(norm, 0);
    }

    public Complex(int norm, int imag) {

        this.norm = norm;
        this.imag = imag;
    }

    public Complex add(Complex second) {

        return new Complex(norm + second.norm, imag + second.imag);
    }

    public Complex add(int n) {

        return new Complex(norm + n, imag);
    }

    public Complex div(Complex second) {
        int den = ((second.norm*second.norm)+(second.imag*second.imag));
        return new Complex(((norm*second.norm) + (imag*second.imag))/den, ((imag*second.norm)-(norm*second.imag))/den);
    }

    public Complex div(int n) {

        return new Complex(norm / n, imag / n);
    }

    public Complex mul(Complex second) {

        return new Complex((norm * second.norm)-(imag * second.imag), (norm * second.imag)+(imag * second.norm));
    }

    public Complex mul(int n) {

        return new Complex(norm * n, imag * n);
    }

    public Complex sub(Complex second) {

        return new Complex(norm - second.norm, imag - second.imag);
    }

    public Complex sub(int n) {

        return new Complex(norm - n, imag);
    }

    public String toString() {

        return "(" + norm + " + " + imag + "i)";
    }

    private int norm;
    private int imag;
}

File: ComplexNumProjC++/complex.cc
#include <iostream>
#include "complex.h"

complex::complex(int norm, int imag) : norm(norm), imag(imag) {}

complex complex::operator+(const complex &second) const {
    return complex(norm + second.norm, imag + second.imag);
}

complex complex::operator+(int n) const {
    return complex(norm + n, imag);
}

complex complex::operator-(const complex &second) const {
    return complex(norm - second.norm, imag - second.imag);
}

complex complex::operator-(int n) const {
    return complex(norm - n, imag);
}

complex complex::operator*(const complex &second) const {
    return complex((norm * second.norm)-(imag * second.imag), (norm * second.imag)+(imag * second.norm));
}

complex complex::operator*(int n) const {
    return complex(norm * n, imag * n);
}

complex complex::operator/(const complex &second) const {
    int den = ((second.norm*second.norm)+(second.imag*second.imag));
    return complex(((norm*second.norm) + (imag*second.imag))/den, ((imag*second.norm)-(norm*second.imag))/den);
}

complex complex::operator/(int n) const {
    return complex(norm / n, imag / n);
}

complex operator+(int n, const complex &second) {
    return second + n;
}

complex operator-(int n, const complex &second) {
    return complex(n) - second;
}

complex operator*(int n, const complex &second) {
    return second * n;
}

complex operator/(int n, const complex &second) {
    return complex(n) / second;
}

ostream &operator<<(ostream &out, const complex &second) {
    out << "(" << second.norm << " + " << second.imag << "i)";
    return out;
}
